### PR TITLE
first_homework

### PR DESCRIPTION
**Pull request by avagyan98**

### CODE REVIEW

File: largestNumber.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+function largest(arr) {
+    return Math.max(...arr)
+}
+
+console.log(largest([3, 5, 1, 37, 26]))
+
+
+function largest2(arr) {
+    arr.sort();
```
Comment (robhov97) [inline largestNumber.js]: the sort should be for specifically numbers

File: largestNumber.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    return arr[-1]
```
Comment (robhov97) [inline largestNumber.js]: it is wrong, try again

File: largestNumber.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+}
+
```
Comment (robhov97) [inline largestNumber.js:27]: do another option with just one standlone loop

File: largestNumber.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+function largest(arr) {
+    return Math.max(...arr)
+}
+
+console.log(largest([3, 5, 1, 37, 26]))
```
Comment (robhov97) [inline largestNumber.js]: don't copy code if you don't understand it

File: product.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
+function product(arr) {
+    let negativeCount = 0;
+    for (let i = 0; i < arr.length; i++) {
+        if (arr[i] === 0) {
+            return 'The product is 0'
+        } else if (arr[i] < 0) {
+            negativeCount ++;
```
Comment (robhov97) [inline product.js]:
```suggestion
            negativeCount++;
```

File: product.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+            continue
+        } else {
+            continue
+        }
```
Comment (robhov97) [inline product.js]: this both continues are redundant

File: product.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
+function product(arr) {
+    let negativeCount = 0;
+    for (let i = 0; i < arr.length; i++) {
+        if (arr[i] === 0) {
+            return 'The product is 0'
+        } else if (arr[i] < 0) {
```
Comment (robhov97) [inline product.js]: else if is redundant

**Review by robhov97 — CHANGES_REQUESTED** [review]

File: largestNumber.js (hunk 4 above)
Comment (avagyan98) [reply largestNumber.js]: done

File: product.js (hunk 7 above)
Comment (avagyan98) [reply product.js]: done

File: product.js (hunk 6 above)
Comment (avagyan98) [reply product.js]: ok) done

File: product.js (hunk 5 above)
Comment (avagyan98) [reply product.js]: done

File: largestNumber.js (hunk 3 above)
Comment (avagyan98) [reply largestNumber.js:27]: done

File: largestNumber.js (hunk 2 above)
Comment (avagyan98) [reply largestNumber.js]: done

File: largestNumber.js (hunk 1 above)
Comment (avagyan98) [reply largestNumber.js]: done)

File: largestNumber.js [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+function largest2(arr) {
+    arr.sort((a, b) => a - b);
+    return arr[arr.length - 1]
+}
+
+console.log(largest2(arr))
+
+
+function largest3(arr) {
+    maxNumber = arr[0];
```
Comment (robhov97) [inline largestNumber.js]: write proper variable declaration

File: product.js [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+    for (let i = 0; i < arr.length; i++) {
+        if (arr[i] < 0) {
+            negativeCount++;
+        }
+    }
+    if (negativeCount % 2 === 0) {
+        return 'The product is positive'
+    }else {
+        return 'The product is negative'
+    }
```
Comment (robhov97) [inline product.js]: else is redundant

File: product.js [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+    console.log(sum)
+
+    if(sum > 0) {
+        return 'The product is positive'
+    }else if(sum < 0) {
+        return 'The product is negative'
+    }else{
+        return 'The product is 0';
+    }
```
Comment (robhov97) [inline product.js]: elses are redundant

**Review by robhov97 — CHANGES_REQUESTED** [review]

File: largestNumber.js (hunk 8 above)
Comment (avagyan98) [reply largestNumber.js]: done

File: product.js (hunk 9 above)
Comment (avagyan98) [reply product.js]: done

File: product.js (hunk 10 above)
Comment (avagyan98) [reply product.js]: done

File: product.js [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+    for (let i = 0; i < arr.length; i++) {
+        if (arr[i] < 0) {
+            negativeCount++;
+        }
+    }
+    if (negativeCount % 2 === 0) {
+        return 'The product is positive'
+    }else {
+        return 'The product is negative'
+    }
```
Comment (robhov97) [inline product.js]: this one still remains

Comment (robhov97) [reply product.js]: It'll be better if you try to convert it to ternary

File: product.js [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+console.log(productMultiply(arr));
+
+
+function productAdd(arr) {
+    let sum = 0;
+
+    for(let i = 0; i < arr.length; i++) {
+        sum += arr[i];
+    }
+    console.log(sum)
```
Comment (robhov97) [inline product.js]: remove log

**Review by robhov97 — CHANGES_REQUESTED** [review]

File: product.js (hunk 11 above)
Comment (avagyan98) [reply product.js]: done

File: product.js (hunk 12 above)
Comment (avagyan98) [reply product.js]: done

File: product.js [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+
+    for(let i = 0; i < arr.length; i++) {
+        sum += arr[i];
+    }
+
+    if(sum > 0) {
+        return 'The product is positive'
+    }
+
+    sum < 0 ? return 'The product is negative' : return 'The product is 0';
```
Comment (robhov97) [inline product.js]: wrong!!!
```suggestion
    return sum < 0 ? 'The product is negative' : 'The product is 0';
```

File: product.js [hunk 14]
```diff
@@ -0,0 +1,40 @@
...
+        if (arr[i] < 0) {
+            negativeCount++;
+        }
+    }
+
+    if (negativeCount % 2 === 0) {
+        return 'The product is positive'
+    }
+
+    return 'The product is negative'
```
Comment (robhov97) [inline product.js]: convert into ternary

File: product.js [hunk 15]
```diff
@@ -0,0 +1,40 @@
...
+
+    if(arr.includes(0)) return 'The product is 0';
+
+    for (let i = 0; i < arr.length; i++) {
+        if (arr[i] < 0) {
+            negativeCount++;
+        }
+    }
+
+    if (negativeCount % 2 === 0) {
```
Comment (robhov97) [inline product.js]: 0 is already a falsy value
```suggestion
    if (!(negativeCount % 2)) {
```

**Review by robhov97 — CHANGES_REQUESTED** [review]

File: product.js (hunk 14 above)
Comment (avagyan98) [reply product.js]: done)

File: product.js (hunk 15 above)
Comment (avagyan98) [reply product.js]: done)

**Review by robhov97 — APPROVED** [review]

**Pull request merged**